src/towncrier/_remover.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) Amber Brown, 2015
+# See LICENSE for details.
+
+from __future__ import annotations
+
+import click
+
+from towncrier._git import remove_files
+
+
+def remove_news_fragment_files(
+    fragment_filenames: list[str], answer_yes: bool, answer_keep: bool
+) -> None:
```

Comment:
Close, but not quite what I had in mind.

We haven't really achieved anything by this refactoring, because this function still deletes the files which makes it an action with pretty heavy side-effects.

My suggestion was that this function returns the `fragment_filenames` as a list and the caller calls `_git.remove_files()` on it unconditionally.

---

Secondarily, you've moved this function into a separate module, so you can easily monkeypatch it without affecting the CLI entry point – correct?

That's pretty heavy "test damage".

How about going a bit functional here:

```python
_git.remove_files(
    find_news_fragment_files_for_removal(  # <-- better name for remove_news_fragment_files after changes
        fragment_filenames, answer_yes, answer_keep,
        confirm_fn=partial(click.confirm, "Is it okay if I remove those files?", default=True)
    )
)
```

In the test you would simply pass `confirm_fn=lambda: True` or `confirm_fn=lambda: False` – look ma no patching!

I would usually not suggest something like that on a public repo, but if the alternative is adding a new module for just one function that is called in one place, it seems worth it – would you agree?

Reply:
Hmmm.... you're right. I got lost a bit between reading your suggestion and sitting down to implement it.

---

When I look at it, the intermediate function only makes a decision on whether to remove the files or not. And would return the fragments_list or not. 
I wonder if it wouldn't be even better if we had something like this:

```python
if should_remove_news_fragment(fragment_filenames, answer_yes, answer_keep):
    _git.remove_files(fragment_files)
```

Then both functions here could be tested separately. And... still, the should_remove_news_fragments could receive confirm_fn partial...

---

Sorry, I tend to split functions sooner, especially if the original module is already several hundred lines big. I'll move it back since... you're right, it's not called anywhere else, and that's a perfect rule to keep 🤔 .

Reply:
@hynek I don't think the functional approach will help in regards to the changed test.

The test that got the mock updated is an integration one (tests from all the way up by calling click command) and... I moved the click calls from `_git.remove` so unless I'll keep the intermediate within `_git` module, you'd have to change the mock anyway 🤔 

And to actually make use of the functional approach the test itself would have to be a unit one.
I'm not sure how to approach it.
And the last question, shall I move the intermediate function into the build module?

Reply:
> Sorry, I tend to split functions sooner, especially if the original module is already several hundred lines big.

yeah the CLI entry points need to be refactored HARD, but let's not blow this up even further 😇

---

since the list is never changed, agreed on `should_remove_news_fragments`

---

ok this click.confirm business is annoying. I was gonna suggest an abstraction, but I've noticed that it's literally only used in that one place so I'm gonna say to KISS and patch away on build.py without the confirm callable, unless you'd like to add more *unit*tests.

Reply:
> but let's not blow this up even further

whew 😌 

---
> unless you'd like to add more unittests

I can't think of any that the integration tests wouldn't already cover... so not at this time. 
I'll try to remember next time I make a contribution though :)